talk/morelanguage/raii.tex

```diff
@@ -47,16 +47,19 @@
   \frametitlecpp[11]{This problem exists for any resource}
   \begin{exampleblock}{For example with a file}
     \begin{cppcode*}{}
-      try {
-        FILE *handle = std::fopen(path, "w+");
-        if (nullptr == handle) { throw ... }
-        if (std::fputs(str, handle) == EOF) {
-          throw ...
-        }
-        fclose(handle);
-      } catch (...) { ... }
+      std::FILE *handle = std::fopen(path, "w+");
+      if (nullptr == handle) { throw ... }
```

Comment:
```suggestion
      if (handle == nullptr) { throw ... }
```
I don't insist, but isn't this easier to read?

Reply:
There is this clan of C++ developers that believes putting an rvalue on the LHS of an equality comparison detects accidental misspelling the operator as assignment:
```c++
if (nullptr = handle) // compiler error
if (handle = nullptr) // compiles fine (but may warn)
```
I think recent compilers warn on this anyway and you should put parentheses around it to silence the warnings:
```c++
if (handle = nullptr) // may warn
if ((handle = nullptr)) // does not warn
```

Reply:
Yes. This is the origin of this. An LHCb old convention, put in place 15-20 years before compilers added the warnings :-) Now I'm not sure it's really useful nowadays and this example should not be polluted by such considerations, so I woudl keep the change